ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -51,16 +60,31 @@ const ApproveStaticSimulation = () => {
           marginInlineEnd={1}
           minWidth={BlockSize.Zero}
         >
-          <Text
-            data-testid="simulation-token-value"
-            backgroundColor={BackgroundColor.backgroundAlternative}
-            borderRadius={BorderRadius.XL}
-            paddingInline={2}
-            textAlign={TextAlign.Center}
-            alignItems={AlignItems.center}
-          >
-            {tokenId}
-          </Text>
+          {tokenAmount === t('unlimited') ? (
+            <Tooltip title={formattedTokenNum}>
+              <Text
+                data-testid="simulation-token-value"
+                backgroundColor={BackgroundColor.backgroundAlternative}
+                borderRadius={BorderRadius.XL}
+                paddingInline={2}
+                textAlign={TextAlign.Center}
+                alignItems={AlignItems.center}
+              >
+                {tokenAmount}
+              </Text>
+            </Tooltip>
+          ) : (
+            <Text
```

Comment:
Are both of these `Text` branches identical? Could we move the ternary inside it?